### ISSUE TITLE
Container shadows are wrong size with `tiny-skia` and custom theme

### ISSUE THREAD

**Issue by mtkennerly** [iced-rs/iced#2339]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Is this issue related to iced?

- [X] My hardware is compatible and my graphics drivers are up-to-date.

### What happened?

I have a custom theme where the `Container` appearance includes:

```
            shadow: Shadow {
                color: Color::BLACK,
                offset: Vector::ZERO,
                blur_radius: 0.0,
            }
```

With `wgpu`, it looks fine (no visible shadow). With `tiny-skia`, it looks incorrect (black fills whole container).

I can work around it by using `Color::TRANSPARENT`.

Gist (using red shadow for illustration purposes): https://gist.github.com/mtkennerly/993c3fbdd7ea770c0c28d268cf063664

### `ICED_BACKEND=wgpu cargo run`
> ![Screenshot 2024-03-18 234327](https://github.com/iced-rs/iced/assets/4934192/98f4492e-5cdb-4436-af1b-5361f705423f)

### `ICED_BACKEND=tiny-skia cargo run`
> ![Screenshot 2024-03-18 234342](https://github.com/iced-rs/iced/assets/4934192/c4e3d979-7fca-4d47-97b0-1fa956710ff0)

### What is the expected behavior?

The `tiny-skia` example should look the same as the `wgpu` example.

### Version

crates.io release

### Operating System

Windows

**Comment by njust** [comment]
Hi, I have the same Issue, but wanted to add, that for buttons shadows work fine also with the tiny-skia backend:

Wgpu backend
![image](https://github.com/iced-rs/iced/assets/2380253/62bc4bf8-9779-4ab4-a8d2-63931eb3c0e6)

Tiny-skia backend
![image](https://github.com/iced-rs/iced/assets/2380253/436f7fcb-a57a-4d93-9e7e-b8957706272c)

What do you mean by "can work around it by using Color::TRANSPARENT."? Using Color::TRANSPARENT for the shadow effectively means no visible shadow. So the workaround for the tiny-skia backend is not using shadows at all?

**Comment by mtkennerly** [comment]
> So the workaround for the tiny-skia backend is not using shadows at all?

My use case is without any shadows, but yeah, that doesn't help if you actually want them 😅

**Comment by SzilvasiPeter** [comment]
The problem is present not just for custom themes, but for the built-in themes as well. The `tiny-skia` backend doesn't draw the style correctly. For demonstration, here is a simple example application that toggle the themes:

```rust
use iced::widget::{button, center, column, text};
use iced::{Element, Theme};

fn main() -> iced::Result {
    iced::application("MyApp", MyApp::update, MyApp::view)
        .theme(MyApp::theme)
        .run()
}

#[derive(Default)]
struct MyApp {
    theme: Theme,
}

#[derive(Debug, Clone)]
enum Message {
    ToggleTheme,
}

impl MyApp {
    fn update(&mut self, message: Message) {
        match message {
            Message::ToggleTheme => {
                if self.theme == Theme::Dark {
                    self.theme = Theme::Light;
                } else {
                    self.theme = Theme::Dark;
                }
            }
        }
    }

    fn view(&self) -> Element<Message> {
        let text = text!("Hello world!");
        let toggle_button = button("Toggle theme").on_press(Message::ToggleTheme);
        let content = column![text, toggle_button].spacing(20);
        center(content).into()
    }

    fn theme(&self) -> Theme {
        self.theme.clone()
    }
}
```

If I run with the `tiny-skia` backend using this command:

```bash
ICED_BACKEND="tiny-skia" cargo run
```

Then, it flickers and incorrectly draw the theme when hovering the mouse over the button.

**EDIT:**

I'd like to upload a GIF to showcase the problem, but I have to switch from Wayland to X11, because [t-rec](https://github.com/sassman/t-rec-rs) supports only X11. Surprisingly, the flickering is gone and the theme switching work with both backend:

![Image](https://github.com/user-attachments/assets/7532f5fb-bf04-4f40-9d83-1b384ac35d9d)

Therefore, the inconsistency issue comes from the differences between Wayland and X11 protocol.

**Comment by T-256** [comment]
I think it solved now on main. Can you test that?

**Comment by d-itu** [comment]
`tiny-skia` shadow is still not correct after c2646f6.

**Comment by T-256** [comment]
> `tiny-skia` shadow is still not correct after [c2646f6](https://github.com/iced-rs/iced/commit/c2646f67a1e342c8367a85969c5ca917c14735c8).

Yes, unfortunately https://github.com/iced-rs/iced/pull/2715#issuecomment-3816680851 still didn't come to main yet.
